examples/gazebo_differential_drive_plugin/differential_drive_plugin.cc

```diff
@@ -0,0 +1,110 @@
+#ifndef _DIFFERENTIAL_DRIVE_PLUGIN_HH_
+#define _DIFFERENTIAL_DRIVE_PLUGIN_HH_
+
+#include <gazebo/gazebo.hh>
+#include <gazebo/physics/physics.hh>
+#include <gazebo/transport/transport.hh>
+#include <gazebo/msgs/msgs.hh>
+
+namespace gazebo
+{
+  /// \brief A plugin to control a simple cart with differential drive.
+  class DifferentialDrivePlugin : public ModelPlugin
+  {
+    /// \brief Constructor
+    public: DifferentialDrivePlugin() {}
+
+    /// \brief The load function is called by Gazebo when the plugin is
+    /// inserted into simulation
+    /// \param[in] _model A pointer to the model that this plugin is
+    /// attached to.
+    /// \param[in] _sdf A pointer to the plugin's SDF element.
+    public: virtual void Load(physics::ModelPtr _model, sdf::ElementPtr _sdf)
+    {
+      // Safety check
+      if (_model->GetJointCount() == 0)
+      {
+        std::cerr << "Invalid joint count, DifferentialDrivePlugin not loaded\n";
+        return;
```

Comment:
I don't know Gazebo that well, so I'm not sure if this is standard practice or not, but I'd suggest in this case that you throw an exception (e.g. ``runtime_error``) rather than just printing a message, as it is essentially a fatal error